torchao/kv_cache.py

```diff
@@ -0,0 +1,407 @@
+import torch
+import torch.nn as nn
+import functools
+from typing import List, Tuple, Optional, Dict, Any
+import copy
+
+HANDLED_FUNCTIONS = {}
+
+
+class PagedTensor(object):
+    def __init__(
+        self,
+        cache: torch.Tensor, #The cache tensor from the PagedAttentionCache object, which is shared accross iterations.
+        block_tables: torch.Tensor,#The block tables for each sequence in the batch which is used to mapping logical block to physical blocks.
+        context_lens: torch.Tensor,#The context lens for each sequence in the batch.
```

Comment:
context length is a concept of text generation, seems not generic to describe a tensor?

Reply:
We have used _size_ which represent the real cache tensor size(bs, num_key_value_heads, seq_lens, head_dim) like the dynamic cache to replace context.